java/src/jmri/jmrit/display/layoutEditor/LayoutEditor.java

```diff
@@ -5312,62 +5314,64 @@ public void mousePressed(MouseEvent event) {
         thisPanel.requestFocusInWindow();
     }   //mousePressed
 
-    private void checkControls(boolean useRectangles) {
-        //check if mouse is on a turnout
-        for (LayoutTurnout t : turnoutList) {
-            selectedPointType = t.findHitPointType(dLoc, useRectangles);
-            if (LayoutTrack.NONE != selectedPointType) {
-                selectedObject = t;
+    // this is a method to iterate over a list of items
+    // calling the predicate tester.test on each one
+    private static boolean forEachItemInListDo(
```

Comment:
Why are you writing your own replacement for the filter() method on Streams here?